python/lsst/drp/tasks/reprocess_visit_image.py

```diff
@@ -0,0 +1,498 @@
+        self.makeSubtask("post_calculations", schema=source_schema)
+
+        source_schema.addField(
+            "visit",
+            type="I",
+            doc="Visit this source appeared on.",
+        )
+        source_schema.addField(
+            "detector",
+            type="U",
+            doc="Detector this source appeared on.",
+        )
+
+        # These fields will be propagated from finalizeCharacterization.
+        # TODO: get them from the finalized catalog instead, so the docstrings
+        # exactly match!
+        source_schema.addField(
+            "calib_psf_candidate",
+            type="Flag",
+            doc="Set if the source was a candidate for PSF determination, "
+            "as determined from FinalizeCharacterizationTask.",
+        )
+        source_schema.addField(
+            "calib_psf_reserved",
+            type="Flag",
+            doc="set if source was reserved from PSF determination by FinalizeCharacterizationTask.",
+        )
+        source_schema.addField(
+            "calib_psf_used",
+            type="Flag",
+            doc="Set if source was used in the PSF determination by FinalizeCharacterizationTask.",
+        )
+
+        # These fields are only here to satisfy the SDM schema, and will
+        # be removed from there as they are misleading (because we don't
+        # propagate this information from gbdes/fgcmcal).
+        source_schema.addField(
+            "calib_photometry_used",
+            type="Flag",
+            doc="Unused; placeholder for SDM schemas.",
+        )
+        source_schema.addField(
+            "calib_photometry_reserved",
+            type="Flag",
+            doc="Unused; placeholder for SDM schemas.",
+        )
+        source_schema.addField(
+            "calib_astrometry_used",
+            type="Flag",
+            doc="Unused; placeholder for SDM schemas.",
+        )
+        source_schema.addField(
+            "calib_astrometry_reserved",
+            type="Flag",
+            doc="Unused; placeholder for SDM schemas.",
+        )
+
+        self.source_schema = afwTable.SourceCatalog(source_schema)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        id_generator = self.config.id_generator.apply(butlerQC.quantum.dataId)
+
+        detector = outputRefs.exposure.dataId["detector"]
+        exposures = inputs.pop("exposures")
+        visit_summary = inputs.pop("visit_summary")
+        background_1 = inputs.pop("background_1")
+        if self.config.do_use_sky_corr:
+            background_2 = inputs.pop("background_2")
+            background = combine_backgrounds(background_1, background_2)
+        else:
+            background = background_1
+
+        # This should not happen with a properly configured execution context.
+        assert not inputs, "runQuantum got more inputs than expected"
+
+        detector_summary = visit_summary.find(detector)
```

Comment:
I think you need to check if this returns nothing (and raise if the detector isn't in the visit_summary).

Reply:
Thanks, I made this a `NoWorkFound`, which I think is what DRP wants in that situation?